spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -123,14 +131,21 @@ else if (payload instanceof ByteString) {
 		}
 	}
 
-	public MessageConverter getMessageConverter() {
-		return this.messageConverter;
+	public Converter<Object, byte[]> getPayloadConverter() {
+		return this.payloadConverter;
 	}
 
-	public void setMessageConverter(MessageConverter messageConverter) {
-		Assert.notNull(messageConverter,
-				"The specified message converter can't be null.");
-		this.messageConverter = messageConverter;
+	/**
+	 * Set the {@link Converter} to convert an outgoing message payload to a {@code byte[]}
+	 * payload for sending to Pub/Sub. If the payload is already of type {@link PubsubMessage},
+	 * {@code byte[]} or {@link ByteString}, the converter is ignored.
+	 * @param payloadConverter converts {@link Message} to a payload of the outgoing message
+	 * to Pub/Sub. Cannot be set to null.
+	 */
+	public void setPayloadConverter(Converter<Object, byte[]> payloadConverter) {
```

Comment:
Right, or just let's not introduce an new ctor since this option is err... optional.